cylc/flow/task_pool.py

```diff
@@ -1696,6 +1704,17 @@ def spawn_task(
         if itask is None:
             return None
 
+        if (
+            spawned_but_not_submitted and
+            not itask.state.outputs.get_completed_outputs()
+        ):
+            # spawned but never submitted, and has no completed outputs.
+            # This implies it was removed, rather than manually set.
+            # TODO: Detecting removal after completion of some outputs probably
+            # TODO: requires recording removal in the DB (set :expired maybe?).
```

Comment:
Beyond the scope of this PR. I'll post a note on Element, then raise an issue.

Reply:
Could you stick up the issue and add a link to it in this comment so we can find our way back to the context when needed.

Note, the `cylc remove` work (will hopefully arrive in 8.4.0) may change this as removed tasks will be bumped into the `None` flow.

Reply:
Actually I've just removed the TODOs. This needs a bit of discussion first - see my post on Element.

Note this bit of code is designed to avoid respawning a previously-removed task (see #6066) - which is at odds with the proposed future of `cylc remove`.  (If erasing flow history, you *want* to be able to respawn the removed tasks, not avoid it).